<launch>

    <arg name="stereo_name" default="stereo"/>
    <arg name="left_camera_name" default="left"/>
    <arg name="right_camera_name" default="right"/>

    <arg name="left_device_name" default="/dev/v4l/by-id/usb-The_Imaging_Source_Europe_GmbH_DFM_22BUC03-ML_03610453-video-index0"/>
    <arg name="right_device_name" default="/dev/v4l/by-id/usb-The_Imaging_Source_Europe_GmbH_DFM_22BUC03-ML_03610446-video-index0"/>

    <arg name="left_camera_info_url" default=""/>
    <arg name="right_camera_info_url" default=""/>

    <arg name="fps" default="30"/>
    <arg name="width" default="744"/>
    <arg name="height" default="480"/>
    <arg name="image_format" default="bayer_grbg8"/>

    <!-- stereo image view -->
    <arg name="stereo_view_enabled" default="false"/>
    <arg name="stereo_view_topic" default="image_rect"/>

    <!-- stereo image processing -->
    <arg name="stereo_proc_enabled" default="true"/>

    <group ns="$(arg stereo_name)">

        <!--left camera -->
        <group ns="$(arg left_camera_name)">

            <!-- video streaming node -->
            <node pkg="lirs_ros_video_streaming" type="video_streamer" name="$(arg left_camera_name)_video_streamer">
                <param name="camera_name" type="string" value="$(arg left_camera_name)"/>
                <param name="device_name" type="string" value="$(arg left_device_name)"/>
                <param name="fps" type="int" value="$(arg fps)"/>
                <param name="frame_id" type="string" value="$(arg left_camera_name)_camera_link"/>
                <param name="camera_info_url" type="string" value="$(arg left_camera_info_url)"/>
                <param name="width" type="int" value="$(arg width)"/>
                <param name="height" type="int" value="$(arg height)"/>
                <param name="image_format" type="string" value="$(arg image_format)"/>
                <remap from="image" to="image_raw"/>
            </node>
        </group>

        <!--right camera -->
        <group ns="$(arg right_camera_name)">

            <!-- video streaming node -->
            <node pkg="lirs_ros_video_streaming" type="video_streamer" name="$(arg right_camera_name)_video_streamer">
                <param name="camera_name" type="string" value="$(arg right_camera_name)"/>
                <param name="device_name" type="string" value="$(arg right_device_name)"/>
                <param name="fps" type="int" value="$(arg fps)"/>
                <param name="frame_id" type="string" value="$(arg right_camera_name)_camera_link"/>
                <param name="camera_info_url" type="string" value="$(arg right_camera_info_url)"/>
                <param name="width" type="int" value="$(arg width)"/>
                <param name="height" type="int" value="$(arg height)"/>
                <param name="image_format" type="string" value="$(arg image_format)"/>
                <remap from="image" to="image_raw"/>
            </node>
        </group>

        <!-- stereo image processing -->
        <node if="$(arg stereo_proc_enabled)" name="stereo_image_proc" pkg="stereo_image_proc" type="stereo_image_proc">
            <env name="ROS_NAMESPACE" value="$(arg stereo_name)"/>
            <param name="approximate_sync" type="bool" value="true"/>
			<param name="min_disparity" value="24"/> <!-- должен быть >= 1 -->
			<param name="disparity_range" value="96"/>
			<param name="speckle_size" value="1000"/>
        </node>
    </group>

    <!-- stereo image view (start only with stereo_image_proc) -->
    <node if="$(eval stereo_view_enabled and stereo_proc_enabled)" name="image_view" pkg="image_view" type="stereo_view">
        <remap from="image" to="$(arg stereo_view_topic)"/>
        <remap from="stereo" to="$(arg stereo_name)"/>
        <param name="approximate_sync" type="bool" value="true"/>
    </node>

    <!-- transform -->
    <node name="tf" pkg="tf" type="static_transform_publisher"
          args="-0.07 0.109 0 0 0 0 base_link $(arg left_camera_name)_camera_link 50"/>
</launch>
